src/cracking.py

```diff
@@ -0,0 +1,307 @@
+
+## Would you rather I print out the output in hashsearch.py or in here ##
+
+class Searcher:
+
+	def __init__(self, config):
+		self.searchers_offline = [] # Temp disabled until I get the IDs format
+		self.searchers_online = [hashtoolkit(), hashsorg(), LmRainbowTabels(), nitrxgen()] # md5cryp() takes ~ 3 seconds to do
+		self.Hash_input = namedtuple("Hash_input", ["text", "types", "api_keys"])
+		
+		self.perform_search(config)
+	
+	def perform_search(self, config):
+
+		for hash in config["hashes"]:
+
+			hash_ctext = next(iter(hash.keys()))
+			keys = [x.lower() for x in hash[next(iter(hash.keys()))].keys()]
+			supported_searchers = []
+			types = []
+
+			if not config["offline"]:
+				for search in self.searchers_online:
+					for hashtype in keys:
+						if hashtype in search.supports:
+							supported_searchers.append(search)
+							types.append(hashtype)
+							break
+
+			for search in self.searchers_offline:
+				for hashtype in keys:
+					if hashtype in search.supports:
+						supported_searchers.append(search)
+						types.append(hashtype)
+						break
+
+			future = self.Hash_input(hash_ctext, types, config["api_keys"])
+
+			with console.status(f"[bold magenta] Searching [reversed] {hash_ctext}") as status:
+				results = self.threaded_search(future, supported_searchers)
+				for k,v in results.items():
+					console.print(f"[bold blue] {k} : [bold red] {v}")
+
+	def threaded_search(self, future, supported_searchers):
+
+		processes = []
+		results = {}
+
+		with ThreadPoolExecutor(max_workers=4) as executor:
+			for search in supported_searchers:
+				processes.append(executor.submit(self.call_searcher, search, future))
+
+		for d in processes:
+			results.update(d.result())
+
+		return results
+	
+	def call_searcher(self, search, future):
+		try:
+			return {type(search).__name__:search.crack(future)}
+		except Exception as e:
+			logger.warn(f"{type(search).__name__} [{e}]")
+			return {type(search).__name__:False}
+
+
+'''
+class hashtoolkit:
+
+	supports = set(["md5", "ntlm"])
+	moduels = ["cloudscraper", "requests"]
+	offline = False
+
+	def crack(self, hash):
+		try:
+			Scraper = cloudscraper.create_scraper()
+			HTML = Scraper.get(
+				f"https://hashtoolkit.com/decrypt-hash/?hash={hash[0]}", timeout=1
+			).text.splitlines()
+		except:
+			logger.warn("Couldn't connect to hashtoolkit")
+			return False  # >> Remember to log this
+
+		for i in range(len(HTML)):
+			if "/generate-hash/?text=" in HTML[i]:
+
+				output = HTML[i].partition("?text=")[2].split('">')[0]
+
+				if "</a></span>" in output or 'title"' in output:
+					logger.warn("Hashtoolkit couldnt crack hash")
+					return False  # >> And this
+				else:
+					return output
+
+		return False  # >> Dont think this would ever happen but log this
+-- removing? '''
+
+# Bug Fixing - API / Domain is currently down
+class hashsorg:
+
+	supports = set(["md5", "NTLM", "SHA-1"])
+	moduels = ["requests"]
+	offline = False
+
+	# How do we implment the API key?

```

Comment:
It's from the config file, which is called in the main class -- it's a dictionary iirc.

You can see it here:
https://github.com/HashPals/HashSearch/wiki/The-API-return-object
https://github.com/HashPals/HashSearch/wiki/API-Keys